packages/components/src/font-size-picker/index.js

```diff
@@ -21,14 +21,19 @@ function getSelectValueFromFontSize( fontSizes, value ) {
 	return 'normal';
 }
 
-function getSelectOptions( optionsArray ) {
+function getSelectOptions( optionsArray, disableCustomFontSizes ) {
+	if ( ! disableCustomFontSizes ) {
+		optionsArray = [
```

Comment:
I think this is a bit confusing to read, since at first glance it looks like you are mutating the original array, when in fact you are only changing the function-scoped variable.

Also, I haven't tested the code yet, but wouldn't adding the custom value here cause problems with the `map` callback, which tries to get the `size` property of every `option`? In any case, it seems a bit confusing.

Reply:
I don't get what you mean about the mutating part ?
The map function goes over the custom option too, but as the custom size needs to be undefined in `getSelectValueFromFontSize` above, it works. I guess it's not optimal, I would like some suggestion on how to improve this piece of code but trying also to be DRY.

Reply:
Well, you're not suppose to mutate props/state directly in React components, and setting `optionsArray` kind of looks like a mutation of the original value passed to `getSelectOptions`. It's **not** actually a mutation, since you're only replacing the local variable's reference value with a reference to an entirely new array; if you had used something like `optionsArray.push()` that _would_ have mutated the array. I just thought it looked a little confusing on first glance.

Good point about how `getSelectValueFromFontSize` works. On second examination, I guess rewriting `getSelectOptions` without introducing unnecessary repetition would be harder than I thought. Nevermind. :slightly_smiling_face: